"Any item in" block for advancements and related changes (Item conditions 1/3)

Goldorion opened the pull request:
This is the first PR trying to achieve the objective of adding as a standalone output block the item condition that will be used by all triggers.

This PR implements a "Any item in" output block, so items related triggers can now select multiple items for their condition (this works the same way enchantments and effects work in #5779). This will also allow, after this is merged, to get a fully working Item condition output block.

This PR also adds a new distinct (separator) category for blocks that are not JSON triggers. All existing output blocks are now located in this category (meaning mcitem selectors, Effect entry, enchantment entry and Any items of)

To make this new block works correctly without big code changes, I changed the fact that the `""` before and after the item's id are now generated directly inside the MCItem block selector's code template. This way, we don't need to deal with checking if the plugged Blockly block is an item selector or the `Any items in` block.

<img width="2555" height="1380" alt="image" src="https://github.com/user-attachments/assets/a135c7a6-e514-42ee-bba1-1e047b8103ec" />

This PR was split from the main one because it ended up with having way more files than originally expected, adding too much visual complexity to the original PR. As everything made in this PR works correctly alone, this PR was made to be easier (and faster?) for everyone.

----
Changelog:
- Added a new advancement item selector: Any item in this list
- Added a new advancement Blockly category: Components

KlemenDEV commented:
Two line title of PR was killing my OCD XD

KlemenDEV commented on src/main/java/net/mcreator/blockly/data/ToolboxCategory.java:
This whole file is shown as changed, likely line endings have changed

KlemenDEV commented on plugins/mcreator-localization/lang/texts.properties:
```diff
@@ -886,6 +886,8 @@ blockly.category.utility=Utility triggers
 blockly.category.utility.description=These triggers should not be used in combination with others.
 blockly.category.world=World related triggers
+blockly.category.components=Components
+blockly.category.components.description=These blocks are used with triggers and are not specific to one.
```
Description only works with dynamically inserted categories so it can be omited

Goldorion commented:
> Tests are not implemented correctly:
> <img alt="image" width="1800" height="174" src="https://private-user-images.githubusercontent.com/16374228/508595743-067811a9-0923-47bf-add8-240e8ed540fa.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjIwMjczMTgsIm5iZiI6MTc2MjAyNzAxOCwicGF0aCI6Ii8xNjM3NDIyOC81MDg1OTU3NDMtMDY3ODExYTktMDkyMy00N2JmLWFkZDgtMjQwZThlZDU0MGZhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTAxVDE5NTY1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg1ZTVlMWFiODBiZDRmZWIyOTIzYzQ1NjQ1YmVhOWQxZDYyZWM0OTJmZDg2Mzk1YzE0NjU5MDNjNzEwMDJhNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7sIRw0rf4XCJUbg_-X0qZ4cldtY_ialo0rctENbqZGM">
> 
> You still did not implement output type test as asked

I implemented it higher. I added its test when we populate the mc item selectors. I thought it was making more sense to add it there. 

<img width="1325" height="513" alt="image" src="https://github.com/user-attachments/assets/5dc7cf09-c7c3-4093-ac6a-d88c61932345" />

KlemenDEV reviewed (changes requested):
This will break things a bit:

<img width="1097" height="477" alt="image" src="https://github.com/user-attachments/assets/02cda578-3f53-4bb7-b16e-ae3202cc7356" />

I recommend following the path @AlsoSomeoneElse set with feature ME:

<img width="593" height="273" alt="image" src="https://github.com/user-attachments/assets/b1a38aba-5161-4c7a-b64b-003412563c98" />

It helps to have uniformized UX across the software + resolved this issue I highlighted.

@AlsoSomeoneElse may be able to help you if you need details of implementation
